solvation_analysis/solution.py

```diff
@@ -431,3 +432,45 @@ def _df_to_atom_group(self, df, solute_index=None):
         if solute_index is not None:
             atoms = atoms | self.solute[solute_index]
         return atoms
+
+    def save_json(self, path):
+        data = {
+            'radii': self.radii,
+            'n_frames': self.n_frames,
+            'n_solute': self.n_solute,
+            'solvent_counts': self.solvent_counts,
+            'solvation_data': self.solvation_data.to_json(),
+            'speciation': self.speciation._as_dict(),
+            'coordination': self.coordination._as_dict(),
+            'pairing': self.pairing._as_dict(),
+            'kernel_kwargs': self.kernel_kwargs,
+            'rdf_init_kwargs': self.rdf_init_kwargs,
+            'rdf_run_kwargs': self.rdf_run_kwargs
+        }
+        with open(path, "w") as f:
+            json.dump(data, f)
+
+    @staticmethod
+    def load_json(path):
+        with open(path) as f:
+            solution_dict = json.load(f)
+        data = {
+            'radii': solution_dict['radii'],
+            'n_frames': solution_dict['n_frames'],
+            'n_solute': solution_dict['n_solute'],
+            'solvent_counts': solution_dict['solvent_counts'],
+            'solvation_data': pd.read_json(solution_dict['solvation_data']),
+            'speciation': Speciation._load_dict(solution_dict['speciation']),
+            'coordination': Coordination._load_dict(solution_dict['coordination']),
+            'pairing': Pairing._load_dict(solution_dict['pairing']),
+            'kernel_kwargs': solution_dict['kernel_kwargs'],
+            'rdf_init_kwargs': solution_dict['rdf_init_kwargs'],
+            'rdf_run_kwargs': solution_dict['rdf_run_kwargs'],
+        }
+        return data
+
+    def as_dict(self):
```

Comment:
This seems quite indirect -- why not have a `to_dict`, `from_dict` akin to a `__setstate__`, `__getstate__` method? 

`save_json` is also not my favourite method name. `to_json` could return a json string perhaps (`json.dumps(obj.to_dict())`), which the user could be responsible for writing to a file, or a `to/from_file` method could handle that.

With a bunch of common mis-matching APIs already out there (e.g.  `pandas.to_json`, `pandas.read_json`; pydantic's `.json()`, `.parse_raw`/ `.parse_file`) keeping a neat `to/from` naming scheme is really valuable.